Orange/widgets/data/owfile.py

```diff
@@ -249,130 +243,143 @@ def browse_file(self, in_demos=False):
+            self.reader.select_sheet(None)
             return
 
-        domain = data.domain
+        self.sheet_combo.clear()
+        self.sheet_combo.addItems(self.reader.sheets)
+        self._select_active_sheet()
+        self.sheet_box.show()
+
+    def _select_active_sheet(self):
+        if self.reader.sheet:
+            try:
+                idx = self.reader.sheets.index(self.reader.sheet)
+                self.sheet_combo.setCurrentIndex(idx)
+            except ValueError:
+                # Requested sheet does not exist in this file
+                self.reader.select_sheet(None)
+        else:
+            self.sheet_combo.setCurrentIndex(0)
+
+    def _describe(self, table):
+        domain = table.domain
         text = "{} instance(s), {} feature(s), {} meta attribute(s)".format(
-            len(data), len(domain.attributes), len(domain.metas))
+            len(table), len(domain.attributes), len(domain.metas))
         if domain.has_continuous_class:
             text += "\nRegression; numerical class."
         elif domain.has_discrete_class:
             text += "\nClassification; discrete class with {} values.".format(
                 len(domain.class_var.values))
-        elif data.domain.class_vars:
+        elif table.domain.class_vars:
             text += "\nMulti-target; {} target variables.".format(
-                len(data.domain.class_vars))
+                len(table.domain.class_vars))
         else:
             text += "\nData has no target variable."
-        if 'Timestamp' in data.domain:
+        if 'Timestamp' in table.domain:
             # Google Forms uses this header to timestamp responses
             text += '\n\nFirst entry: {}\nLast entry: {}'.format(
-                data[0, 'Timestamp'], data[-1, 'Timestamp'])
-        self.info.setText(text)
-
-        add_origin(data, self.loaded_file)
-        self.send("Data", data)
+                table[0, 'Timestamp'], table[-1, 'Timestamp'])
+        return text
+
+    def storeSpecificSettings(self):
+        self.current_context.modified_variables = self.variables[:]
+
+    def retrieveSpecificSettings(self):
+        if hasattr(self.current_context, "modified_variables"):
+            self.variables[:] = self.current_context.modified_variables
+
+    def apply_domain_edit(self):
+        attributes = []
+        class_vars = []
+        metas = []
+        places = [attributes, class_vars, metas]
+        X, y, m = [], [], []
+        cols = [X, y, m]  # Xcols, Ycols, Mcols
+        for column, (name, tpe, place, vals, is_con), (orig_var, orig_plc) in \
+            zip(count(), self.editor_model.variables,
+                chain([(at, 0) for at in self.data.domain.attributes],
+                      [(cl, 1) for cl in self.data.domain.class_vars],
+                      [(mt, 2) for mt in self.data.domain.metas])):
+            if place == 3:
+                continue
+            col_data = list(chain(*self.data[:, orig_var]))
```

Comment:
Should this be inside some if/elif/else also?
Otherwise it seems like it is computed here and overwritten 2 lines below (if orig_plc==2).
(At first glance, at least it can't hurt to put it in an else of the next if, possibly something even better)